appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -197,12 +201,26 @@ internal class ExperienceRenderer(
         stateMachines.getOwner(renderContext)?.stateMachine?.handleAction(StartStep(stepReference))
     }
 
-    suspend fun preview(experienceId: String): Boolean {
-        repository.getExperiencePreview(experienceId)?.let {
-            return show(it)
-        }
+    sealed class PreviewResponse {
+        data class Success(val experience: Experience) : PreviewResponse()
+        data class NetworkError(val throwable: Throwable?) : PreviewResponse()
+        data class HttpError(val code: Int?) : PreviewResponse()
+        data class RenderDeferred(val frameId: String, val name: String) : PreviewResponse()
+    }
 
-        return false
+    suspend fun preview(experienceId: String): PreviewResponse {
+        repository.getExperiencePreview(experienceId).run {
+            if (this is PreviewResponse.Success) {
+                val renderContext = experience.renderContext
+                previewExperiences[renderContext] = experience
+
+                if (!show(experience) && renderContext is RenderContext.Embed) {
```

Comment:
I think ideally `show()` is updated to return a result with success/failure, and only return the `RenderDeferred` error in the specific case that occurs L94. Then, any other failure would be properly handled. There could be some other reason an embed failed to render, in theory (maybe not today), so `show()` would need to pass that information back.